x-pack/platform/plugins/shared/lens/public/editor_frame_service/editor_frame/state_helpers.ts

```diff
@@ -306,8 +312,11 @@ export function initializeVisualization({
       visualizationMap[visualizationState.activeId]?.initialize(
         () => '',
         visualizationState.state,
-        // initialize a new visualization with the color mapping off
         COLORING_METHOD,
+        'formBased' in datasourceStates
```

Comment:
The `formBased` key may be there for other reasons so better check the actual state:

```suggestion
        'formBased' in datasourceStates && datasourceStates.formBased.state
```